pkg/version/constraint.go

```diff
@@ -90,15 +102,43 @@ func NewConstraints(v string) (Constraints, error) {
 
 }
 
+func rewriteRange(i string) string {
+	m := constraintRangeRegexp.FindAllStringSubmatch(i, -1)
+	if m == nil {
+		return i
+	}
+	o := i
+	for _, v := range m {
+		t := fmt.Sprintf(">= %s, <= %s.*", v[1], v[11])
+		o = strings.Replace(o, v[0], t, 1)
+	}
+	return o
+}
+
 func newConstraint(c string) (Constraint, error) {
+	if c == "" {
+		return Constraint{
+			version: Version{
+				segments: part.NewParts("*"),
+			},
+			operatorFunc: constraintOperators[""],
+		}, nil
+	}
+
 	m := constraintRegexp.FindStringSubmatch(c)
 	if m == nil {
 		return Constraint{}, xerrors.Errorf("improper constraint: %s", c)
 	}
 
-	v, err := Parse(m[2])
-	if err != nil {
-		return Constraint{}, xerrors.Errorf("version parse error (%s): %w", m[2], err)
+	var segments []part.Part
+	for _, str := range strings.Split(m[3], ".") {
+		segments = append(segments, part.NewPart(str))
+	}
+
+	v := Version{
+		segments:   segments,
+		preRelease: part.NewParts(m[6]),
+		original:   c,
```

Comment:
I'm curious why `Parse` doesn't fit here.

Reply:
If we use `Parse` from `version.go` here that means a valid version would be `1.2.x`. So instead it's necessary to parse the segments here to create a valid "constraint" version.